site/content/blog/2020-12-01-whats-new-in-svelte-december-2020.md

```diff
@@ -0,0 +1,71 @@
+---
+title: What's new in Svelte: December 2020
+description: Better tooling, export maps and improvements to slots and context
+author: Daniel Sandoval
+authorURL: https://desandoval.net
+---
+
+It's the last "What's new in Svelte" of the year and there's lots to celebrate! This month's coverage includes updates from `rollup-plugin-svelte`, `Sapper` and `SvelteKit` and a bunch of showcases from the Svelte community!
+
+## New features & impactful bug fixes
+
+1. `$$props`, `$$restProps`, and `$$slots` are all now supported in custom web components (**3.29.5**, [Example](https://svelte.dev/repl/ad8e6f39cd20403dacd1be84d71e498d?version=3.29.5)) and  `slot` components now support spread props: `<slot {...foo} />` (**3.30.0**)
+2. A new `hasContext` lifecycle function makes it easy to check whether a `key` has been set in the context of a parent component (**3.30.0** & **3.30.1**, [Docs](https://svelte.dev/docs#hasContext))
+3. `SvelteComponent` is now typed which makes it easier to add typed classes that extend base Svelte Components. Component library and framework authors rejoice! An example: `export class YourComponent extends SvelteComponent<{aProp: boolean}, {click: MouseEvent}, {default: {aSlot: string}}> {}` (**3.30.0**, [RFC](https://github.com/sveltejs/rfcs/pull/37))
+4. Transitions within `{:else}` blocks should now complete successfully (**3.29.5**, [Example](https://svelte.dev/repl/49cef205e5da459594ef2eafcbd41593?version=3.29.5))
+5. Svelte now includes an export map, which explicitly states which files can be imported from its npm package. This should significantly improve debugging in web dev tools (**3.29.5** with some fixes in **3.29.6**, **3.29.7** and **3.30.0**)
```

Comment:
> This should significantly improve debugging in web dev tools

oh cool. I didn't know that. It might be interesting to link this section of text to an article discussing that if one exists

Reply:
I also did not know that / cannot find anything about better debugging in that context. It helps to know what files to retrieve from which paths, and to close off private paths, but I did not find anything about debugging.

```suggestion
5. Svelte now includes an export map, which explicitly states which files can be imported from its npm package (**3.29.5** with some fixes in **3.29.6**, **3.29.7** and **3.30.0**)
```